pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -66,12 +66,16 @@
 
   private final ScheduledExecutorService _executorService;
   private final int _runFrequencyInSeconds;
+  private final int _deletedSegmentsRetentionInDays;
 
   private static final int RETENTION_TIME_FOR_OLD_LLC_SEGMENTS_DAYS = 5;
+  private static final int DEFAULT_RETENTION_FOR_DELETED_SEGMENTS_DAYS = 7;
 
-  public RetentionManager(PinotHelixResourceManager pinotHelixResourceManager, int runFrequencyInSeconds) {
+  public RetentionManager(PinotHelixResourceManager pinotHelixResourceManager, int runFrequencyInSeconds,
```

Comment:
Unsure if it is a good idea to have Retention manager do this cleanup, or should it be a separate service. Can you share why you went with the former?